operator/pkg/apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -214,6 +214,8 @@ type Rack struct {
 	Name string `json:"name"`
 	// Zone name to pin the rack, using node affinity
 	Zone string `json:"zone,omitempty"`
+
+	Partition int32 `json:"partition,omitempty"`
```

Comment:
I don't think we should put this on the rack. It's just a footgun waiting to happen and there's nothing much to be gained by letting it vary per rack. Let's add CanaryUpgradeCount for the number of pods to roll forward, right next to CanaryUpgrade. If it's missing - the default - you get the current behavior for CanaryUpgrade.

Reply:
>I don't think we should put this on the rack. It's just a footgun waiting to happen

Can you expand on why it is a bad idea to put it on the rack?

>there's nothing much to be gained by letting it vary per rack.

Letting vary by rack allows you to upgrade a single node and it also gives you the flexibility to choose the rack. Do we get that with `CanaryUpgradeCount`?